server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileFactory.java

```diff
@@ -60,7 +61,10 @@ RulesProfileDto getOrCreateCustom(DbSession dbSession, OrganizationDto organizat
     RulesProfileDto profile = db.qualityProfileDao().selectByNameAndLanguage(organization, name.getName(), name.getLanguage(), dbSession);
     if (profile == null) {
       profile = doCreate(dbSession, organization, name, false, false);
+    } else {
+      checkArgument(!profile.isBuiltIn(), "Operation forbidden for built-in Quality Profile '%s'", profile.getKey());
```

Comment:
The error message should contain the profile name and language. The key, which is a UUID, is not human-readable.